common/src/main/java/com/microsoft/identity/common/internal/broker/InactiveBrokerClient.java

```diff
@@ -0,0 +1,179 @@
+package com.microsoft.identity.common.internal.broker;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.RemoteException;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import com.microsoft.aad.adal.IBrokerAccountService;
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Use for communicating with inactive broker only.
+ * This is NOT being used in ADAL. If you plan to do so, make sure it works properly.
+ **/
+public class InactiveBrokerClient {
+    private static final String TAG = InactiveBrokerClient.class.getSimpleName();
+
+    private static final String BROKER_ACCOUNT_SERVICE_INTENT_FILTER = "com.microsoft.workaccount.BrokerAccount";
+    private static final String BROKER_ACCOUNT_SERVICE_CLASS_NAME = "com.microsoft.aad.adal.BrokerAccountService";
+
+    private Context mContext;
+    private String mActiveBrokerPackageName;
+    private BrokerAccountServiceConnection mBrokerAccountServiceConnection;
+    private Intent mBrokerAccountServiceIntent;
+    private Boolean mBound = false;
+
+    // Must be called from BG thread.
+    public static String getSerializedSymmetricKeyFromInactiveBroker(@NonNull Context context,
+                                                                     @NonNull final String activeBrokerPackageName) {
+        final String methodName = ":getSerializedSymmetricKeyFromInactiveBroker";
+
+        final InactiveBrokerClient client = new InactiveBrokerClient(context, activeBrokerPackageName);
```

Comment:
Is the ActiveBrokerPackageName ever not the current package name? I think this is based on the design supports push or pull models, so we don't care? Or if we're deciding on pull, we should at least log or throw in the unexpected case of them not matching?

Reply:
It is the current package name. I just expose this (instead of directly pulling it from context), for mock testing).

I could rename it if this is confusing.